src/evm_parse.c

```diff
@@ -34,76 +135,216 @@ enum parse_rv parse_rlp_txn(struct EVM_RLP_list_state *const state, evm_parser_m
+              // In particular, all values are at least plausible here. We could show it perhaps.
+            PARSE_ITEM(EVM_TXN_GASPRICE, _to_buffer);
+            // Probably needs saved and/or prompted here
+            PARSE_ITEM(EVM_TXN_STARTGAS, _to_buffer);
+            // Probably needs saved and/or prompted here
+            PARSE_ITEM(EVM_TXN_TO, _to_buffer);
+
+            if(state->rlpItem_state.length!=20) REJECT("Desination address not 20 bytes");
+
+            for(uint64_t i=0; i<sizeof(precompiled)/sizeof(struct known_destination); i++) {
+                if(!memcmp(precompiled[i].to, state->rlpItem_state.buffer, 20)) {
+                    meta->known_destination = &precompiled[i];
+                    break;
+                }
+            }
+            if(!meta->known_destination) {
+                SET_PROMPT_VALUE(memcpy(entry->data.output_prompt.address.val,state->rlpItem_state.buffer, 20));
             }
 
-            init_rlp_item(&state->rlpItem_state);
+            PARSE_ITEM(EVM_TXN_VALUE, _to_buffer);
+
+            uint64_t value = 0; // FIXME: support bigger numbers.
+            if(state->rlpItem_state.length>6) REJECT("Can't support large numbers (yet)") // Fix this.
+            for(uint64_t i=0;i<state->rlpItem_state.length;i++) // Should be a function.
+                ((uint8_t*)(&value))[i]=state->rlpItem_state.buffer[state->rlpItem_state.length-i];
+
+            SET_PROMPT_VALUE(entry->data.output_prompt.amount = value);
 
-            if(state->remaining == 0) {
-                state->state=3;
+            if(ADD_ACCUM_PROMPT(
+                "Transfer",
+                output_evm_prompt_to_string // Needs tweaked to be EVM addresses
+                )) return PARSE_RV_PROMPT;
 
-                return PARSE_RV_DONE;
+            PARSE_ITEM(EVM_TXN_DATA,);
+
+            if(meta->known_destination) {
+                sub_rv = meta->known_destination->handle_data(&(state->rlpItem_state.endpoint_state), &(state->rlpItem_state.chunk), meta);
+            } else if(state->rlpItem_state.length != 0) {
+                // Probably we have to allow this, as the metamask constraint means _this_ endpoint will be getting stuff it doesn't understand a lot.
+                static char const isPresentLabel[]="Is Present";
+                if(ADD_PROMPT("Contract Data", isPresentLabel, sizeof(isPresentLabel), strcpy_prompt)) return PARSE_RV_PROMPT;
             }
-        }
+
+            PARSE_ITEM(EVM_TXN_CHAINID, _to_buffer);
+
+            if(state->rlpItem_state.length != 2 || state->rlpItem_state.buffer[0] != 0xa8
+              || (state->rlpItem_state.buffer[1] != 0x69 && state->rlpItem_state.buffer[1] != 0x6a))
+                REJECT("Chain ID incorrect for the Avalanche C chain");
+            meta->chainIdLowByte = state->rlpItem_state.buffer[state->rlpItem_state.length-1];
+            meta->chainIdLowByte = meta->input.src[meta->input.consumed-1];
```

Comment:
Two assignments to the same field?

Reply:
mistake, please fix. The first one is grabbing it out of the subparser output, so use that (and fix it if it's not getting the right element).